### ISSUE TITLE
Got 'Failed to get genes: : 400 Bad Request. Please try again later.' while trying to "View in Gene Network"

### ISSUE THREAD

**Issue by mhuyck** [greenelab/adage-server#148]
Steps to reproduce:

1. Search for 'anr' under Analyze tab
2. Experiment `E-GEOD-17296` will be at the top. Add its samples to the sample bin (click `+` at right)
3. Select all 4 samples with genotype "WT" for the base group (purple)
4. Select all 4 samples with genotype "∆anr" for the comparison group (yellow)

![screen shot 2017-05-09 at 9 28 04 am](https://cloud.githubusercontent.com/assets/1151166/25853175/dc52de18-3499-11e7-8cb0-96b3b11e89dc.png)

5. Click "Volcano Plot" button and select "Node41pos" (the node shown below near the upper right of the plot)

![screen shot 2017-05-09 at 9 21 51 am](https://cloud.githubusercontent.com/assets/1151166/25853210/00cf2350-349a-11e7-8548-b6ee1b16a3ba.png)

6. Click the "View in Gene Network" button and the "Failed to get genes" message appears.

**Comment by cgreene** [comment]
URL longer than allowed? Just a guess...

**Comment by mhuyck** [comment]
This is very similar to bug #146, but the error appears in a slightly different place so the message is different.

**Comment by mhuyck** [comment]
Based upon what I see in the console log, @cgreene's hunch appears to be exactly what it is. I suspect the same issue with #146.

**Comment by mhuyck** [comment]
The URL it's trying to get is this, which is 4432 characters long:

```
http://adage.greenelab.com/api/v0/gene/?limit=0&pk__in=270,255,254,628,276,57,3209,3208,3207,1960,1959,4200,4199,4208,4207,3639,476,2835,2834,3524,3518,3431,3391,3365,3407,3355,3843,3861,3827,3924,4238,4237,2207,1392,1391,1384,1090,1262,748,747,2215,2273,2534,2533,1243,1446,272,2124,4903,4875,4842,1284,4994,5121,5132,4738,5279,4585,4600,1043,1010,2276,1180,3077,1697,2037,1688,1764,5443,5460,5447,5403,4210,1858,846,2770,3907,3876,220,4526,921,4583,4589,4620,4623,1879,4601,1955,2176,138,687,3092,1948,2038,1911,1829,4538,5182,4892,4278,4542,4212,2133,2926,4195,2194,1952,2195,2426,4972,4599,3392,1619,1320,4113,5307,3741,4030,4288,478,16,2390,4246,5216,4073,2745,707,5259,5349,3866,3427,1557,3753,833,4163,179,3217,786,268,5348,4603,3305,2325,3272,524,3335,4921,5438,2829,118,2237,5093,1889,4211,4402,1774,2853,4185,5451,4773,2308,1330,2397,4118,4440,1211,4960,1786,1950,4896,5262,1065,381,2113,3672,4203,130,4127,4772,4593,1728,4169,1264,3311,2090,4617,3467,1501,1544,2424,4177,2196,1333,3016,2850,3591,2878,51,2641,4147,1360,5058,3597,1511,627,2986,3086,1616,1014,1676,4079,3758,3284,1326,2041,2741,4596,4048,3238,3512,4068,184,1345,5100,4730,4575,1664,1022,3917,3620,4175,1146,1035,4444,1518,977,3925,5385,2576,4728,4606,5375,1371,1946,105,4373,3717,3702,713,4330,361,2580,4080,1490,4962,4130,1848,2307,1059,735,35,2500,4920,470,1307,3345,677,4125,869,3078,4019,576,161,2030,3803,3036,4340,465,911,4826,269,4940,2212,734,1227,4236,1051,4494,1824,4749,2290,4153,855,4220,4501,1259,2391,3379,756,4255,2085,5199,3393,1296,4095,1805,757,709,3662,1583,4018,292,2238,1533,1346,914,936,2021,1260,4178,186,3336,245,4727,3348,1165,4105,1630,3585,5392,2769,3832,467,3604,804,3155,1308,4137,433,609,2832,4280,2569,2779,1270,3619,632,445,1179,3998,597,4748,4513,2101,5395,4467,1976,1410,3433,4485,1957,3270,4898,3170,4286,1166,779,2114,3046,5196,3910,3171,1691,4781,4111,4158,4157,3416,3132,1523,1009,1875,4919,2200,176,4495,3954,5520,894,3685,4217,2329,4774,4154,3615,1628,3470,2801,4162,3390,3757,2197,2140,1066,3342,4750,4336,3383,3618,1283,2496,1006,935,2265,2115,2540,847,2927,4166,2157,5134,3252,1947,1681,4108,2654,67,4967,359,4366,2125,4462,1457,3100,950,2330,3158,3806,380,1844,3951,2159,4126,4549,2929,343,332,2464,1046,3915,74,2784,2463,4141,4285,4609,3448,173,5071,4221,3010,5271,4701,2128,4901,3281,4090,2069,1980,1361,1388,2322,5314,4636,190,676,3314,4642,5154,499,1001,615,994,2612,3289,4254,2913,922,1433,4693,3659,5492,4361,4225,1255,3469,1661,910,999,1374,4445,2264,1738,2057,1139,3408,1558,1966,5496,1327,72,4463,258,376,3476,4930,621,162,4707,4742,3034,4704,5191,5402,2223,4917,1920,3428,852,1621,4719,5430,2013,4083,3706,4251,222,3511,4554,4436,2718,4046,4422,440,4186,2403,2825,5004,2880,93,4754,3167,181,4480,4170,3796,4902,5485,4188,5378,1297,512,3777,2965,3888,239,4821,527,4258,3878,338,5437,4808,4450,3003,5055,3552,4194,1917,1958,2478,1556,409,5261,3992,282,5185,2957,5401,1723,4335,538,3071,3045,5376,2205,2355,4316,4405,2324,4729,3466,3782,3004,2046,1373,4569,4667,2893,4520,5270,3906,3650,2451,4691,1385,3306,2121,1056,1590,953,2786,3360,342,1993,3734,1500,1231,1582,4489,4243,3285,1629,1502,689,158,3022,3308,3558,76,688,1409,2995,3479,5399,2937,5404,4582,3821,5114,1515,4349,3258,4713,4218,1768,1491,848,3590,3974,4358,3203,4629,3716,3253,253,3356,2864,1007,2921,4973,3514,1843,4426,3966,314,1580,4435,4379,177,3606,2794,4721,4739,201,3066,3598,4528,3778,581,4051,4329,4894,4893,1232,4328,5491,3400,5028,4047,3614,4397,1975,1969,4711,2033,285,2922,456,3458,464,4764,769,4966,1517,5327,3973,1441,932,1376,1241,2764,3145,3665,3921,3495,3903,2993,1248,1344,3862,3421,5386,1375,3531,4104,4229,1602,1134,3575,1252,5247,1729,4879,2795,2131,4638,4895,3772,5176,3975,5495,182,4268,4350,972,754,2925,4282,3605,4202,5324,2555,1520,1840,4224,4059,1387,223,4869,1467,339,3636,89,3339,750,2047,796,2331,3302,585,300,3047,703,4897,695,1253,4160,3354,567,2766,4737,4747,4635,2768,5366,340,3635,2119,4039,2364,2008,2031,4943,274,4119,2166,1129,1004,4900,1127,2888,3532,4390,647,3329,1445,1368,864,3557,2879,3779,5474,4899,1859,1390,2250,3074,2354,961,3083,2975,4049,4439,2305,1675,4556,978,1716,4232,5179,1907,2558,708,4745,659,1967,146,4705,933,2851,372,3804,5278,185,4114,4706,4165,3190,1942,1973,2554,1519,2947,1370,4790,3389,3671,949,1324,3116,5425,82,4610,1857,1992,4003,132,4498,1890,3294,4746,3014,1974,341,875,136,4110,1940,2501,3537,1885,1874,5357,4368,3239,194,3684,259,416,2239,4183,544,218,4348,4932,1622,4012
```

This problem arises here as a result of the way I implemented the fix in #112. We will have to break up this query or make the API accept a `POST` request for the list of gene IDs.

**Comment by dongbohu** [comment]
Thank you for digging deeper on this issue.

**Comment by mhuyck** [comment]
Turns out that the way to fix this will likely be a change in `django-genes`, which is a repo over at Bitbucket. The root issue (and fix) is the same in issue #146, so my next updates will be to that code. Once I have a working fix, will file a PR on `django-genes` as well.

**Comment by mhuyck** [comment]
Fixed in #174.
